module/zfs/dbuf.c

```diff
@@ -860,7 +1008,7 @@ dbuf_evict_notify(uint64_t size)
 	 */
 	if (size > dbuf_cache_target_bytes()) {
 		if (size > dbuf_cache_hiwater_bytes())
-			dbuf_evict_one();
+			dbuf_evict();
```

Comment:
Lets assume we have 10 user threads calling this.  I suppose each of them will try to create own task sets to evict the same full amount of extra dbuf caches using all the same CPUs.  In best case it may end up with empty dbuf cache.  I am not sure I greatly like the design of one main eviction thread calling bunch of other taskqs, but each client thread doing that definitely looks weird.  I think if user threads has to do evictions, they should do it directly, just doing more than one buffer at a time to be more efficient, as you have said.